corehq/apps/cleanup/management/commands/delete_duplicate_users.py

```diff
@@ -0,0 +1,107 @@
+import itertools
+from collections import defaultdict
+
+from django.contrib.auth.models import User
+from django.core.management import BaseCommand
+
+from corehq.apps.es import FormES
+from corehq.apps.users.dbaccessors.all_commcare_users import (
+    get_all_user_id_username_pairs_by_domain,
+)
+from corehq.apps.users.models import CommCareUser, CouchUser
+
+
+class Command(BaseCommand):
+    help = """Find and delete duplicate mobile users in a domain"""
+
+    def add_arguments(self, parser):
+        parser.add_argument('domain')
+
+    def handle(self, domain, **options):
+        duplicate_users = get_duplicate_users(domain)
+
+        if not duplicate_users:
+            print('No duplicates found')
+            return
+
+        print(f'Found {len(duplicate_users)} usernames with duplicates')
+        all_user_id = list(itertools.chain.from_iterable(duplicate_users.values()))
+        user_ids_with_forms = get_users_with_forms(domain, all_user_id)
+
+        for username, user_ids in duplicate_users.items():
+            users_by_id = {
+                user_id: CommCareUser.get_by_user_id(user_id)
+                for user_id in user_ids
+            }
+            safe_user_ids = set(user_ids) - set(user_ids_with_forms)
+
+            if not safe_user_ids:
+                print(f'All users for {username} have forms. Skipping user IDs {user_ids}')
```

Comment:
Should this note any duplicate user ids that do have forms? Aren't those users still a problem?

Reply:
This would happen if all duplicate users have form submissions in which case a more complex solution is required that's beyond the scope of this script.

Reply:
I don't mean to fix those users, just to let the person running the script know they exist.

Reply:
I'm not sure I follow, this does output the user IDs.

Reply:
But it only outputs them if all duplicate users have forms, right? If there are some duplicates that have forms and some that don't, I'd expect the script to delete the users without forms (like it does) but also to output something like "There are another 10 duplicate users that have forms and therefore weren't deleted, ids are: ..."

Reply:
I see - I'll update